src/Temporalio/Bridge/src/metric.rs

```diff
@@ -186,3 +196,295 @@ impl From<&MetricIntegerOptions> for metrics::MetricParameters {
             .unwrap()
     }
 }
+
+type CustomMetricMeterMetricIntegerNewCallback = unsafe extern "C" fn(
+    name: ByteArrayRef,
+    description: ByteArrayRef,
+    unit: ByteArrayRef,
+    kind: MetricIntegerKind,
+) -> *const libc::c_void;
+
+type CustomMetricMeterMetricIntegerFreeCallback = unsafe extern "C" fn(metric: *const libc::c_void);
+
+type CustomMetricMeterMetricIntegerUpdateCallback =
+    unsafe extern "C" fn(metric: *const libc::c_void, value: u64, attributes: *const libc::c_void);
+
+type CustomMetricMeterAttributesNewCallback = unsafe extern "C" fn(
+    append_from: *const libc::c_void,
+    attributes: *const CustomMetricAttribute,
+    attributes_size: libc::size_t,
+) -> *const libc::c_void;
+
+type CustomMetricMeterAttributesFreeCallback =
+    unsafe extern "C" fn(attributes: *const libc::c_void);
+
+type CustomMetricMeterMeterFreeCallback = unsafe extern "C" fn(meter: *const CustomMetricMeter);
+
+/// No parameters in the callbacks below should be assumed to live beyond the
+/// callbacks unless they are pointers to things that were created lang-side
+/// originally. There are no guarantees on which thread these calls may be
+/// invoked on.
+#[repr(C)]
+pub struct CustomMetricMeter {
+    pub metric_integer_new: CustomMetricMeterMetricIntegerNewCallback,
+    pub metric_integer_free: CustomMetricMeterMetricIntegerFreeCallback,
+    pub metric_integer_update: CustomMetricMeterMetricIntegerUpdateCallback,
+    pub attributes_new: CustomMetricMeterAttributesNewCallback,
+    pub attributes_free: CustomMetricMeterAttributesFreeCallback,
+    pub meter_free: CustomMetricMeterMeterFreeCallback,
+}
+
+#[repr(C)]
+pub struct CustomMetricAttribute {
+    pub key: ByteArrayRef,
+    pub value: CustomMetricAttributeValue,
+    pub value_type: MetricAttributeValueType,
+}
+
+#[repr(C)]
+pub union CustomMetricAttributeValue {
```

Comment:
You can, I believe, still represent this as an enum: https://doc.rust-lang.org/reference/type-layout.html#combining-primitive-representations-of-enums-with-fields-and-reprc

Not terribly important though, just wanted to point it out.

Reply:
I think that's only if every value in the enumerate would be the same size/primitive